fix for CI (black)

tomsail opened the pull request:
To solve #77 

I am first putting as draft to if CI passes

krober10nd commented:
Seems like there were some issues with starting up the CI system with versions of Python other than 3.9?

pmav99 commented:
The reason this fails is because you are asking black to automatically fix the issues it finds and it seems that there are permissions issues when it tries to amend the commits. You can either:

1. Try to fix the permissions issues: https://github.com/CHLNDDEV/oceanmesh/actions/runs/8567158490/job/23478362132?pr=78#step:5:89
2. Use a flag like `black --check` or `black --diff` which will fail if there are differences without trying to make changes to the source code. Not sure how you do that with `wearerequired/lint-action@master` but the docs should mention something.

tomsail commented:
So there are various problems here: 

 1. the formatter CI: the [permissions issue](https://github.com/CHLNDDEV/oceanmesh/actions/runs/8567158490/job/23478362132?pr=78#step:5:89) with `wearerequired/lint-action@master` does not work for forks. Issue is still [open](https://github.com/wearerequired/lint-action/issues/13) on `wearerequired` page. We might want to test in a different way. 
 2. `sudo apt install -y libopenmpi3 libopenmpi-dev openmpi-bin` failing for some python versions. example [here](https://github.com/CHLNDDEV/oceanmesh/actions/runs/8567158494/job/23478362334?pr=78)
 3. `black` version issue. I will try to see which version throws the least number of errors. I had started to test (as you can see in the commit), but didn't finish it.

tomsail commented:
> the formatter CI: the [permissions issue](https://github.com/CHLNDDEV/oceanmesh/actions/runs/8567158490/job/23478362132?pr=78#step:5:89) with wearerequired/lint-action@master does not work for forks. Issue is still https://github.com/wearerequired/lint-action/issues/13 on wearerequired page. We might want to test in a different way.

It works anyway if black and flake go through. Although permissions are still an issue (see [last logs](https://github.com/CHLNDDEV/oceanmesh/actions/runs/8603087925/job/23574141270?pr=78)), if linting is fixed it won't raise any error at the end. 

> sudo apt install -y libopenmpi3 libopenmpi-dev openmpi-bin failing for some python versions. example [here](https://github.com/CHLNDDEV/oceanmesh/actions/runs/8567158494/job/23478362334?pr=78)

Fixed by adding "`sudo apt update` in the CI steps"

> black version issue. I will try to see which version throws the least number of errors. I had started to test (as you can see in the commit), but didn't finish it.

Fixed by removing: 
```
line-length = 108
```
and letting default maximum line-length

tomsail commented:
should be all good now. let's wait to see if latest CI pass

krober10nd commented:
Excellent thank you Thomas.

krober10nd commented on .github/workflows/testing.yml:
```diff
@@ -12,7 +12,7 @@ jobs:
     - uses: actions/setup-python@v2
       with:
-        python-version: "3.x"
+        python-version: "3.7"
```
Does the 3.x imply that many versions of Python would be tested? Is it possible to test the last three trailing versions minus the current (which is typically always problematic)?

tomsail replied:
It's a sort of "wildcard" that matches the latest minor python3 version in github CI.
Since @pmav99 suggested to use the lowest available, I changed it to `3.7`

tomsail commented:
@krober10nd is it really necessary to have the `formatter.yml` CI action? 
It seems a bit overkill to use and deploy `wearerequired/lint-action@master` (which is itself a library that can induce changes) for just the linting (that we already have in the test CI action.

krober10nd commented:
it is not essential since the PR author will know if it is violating the format. I'm in favor of removing it.
